pkg/application/application.go

```diff
@@ -41,97 +33,38 @@ func GetDefaultAppName(existingApps []preference.ApplicationInfo) (string, error
 
 	// If there's no prefix in config file or it is equal to $DIR, use safe default which is the name of current directory
 	if cfg.OdoSettings.NamePrefix == nil || *cfg.OdoSettings.NamePrefix == "" {
-		prefix, err := component.GetComponentDir("", occlient.NONE)
+		prefix, err := component.GetComponentDir("", config.NONE)
 		if err != nil {
 			return "", errors.Wrap(err, "unable to generate random app name")
 		}
-		appName, err = util.GetRandomName(prefix, appPrefixMaxLen, existingAppNames, appNameMaxRetries)
+		appName, err = util.GetRandomName(prefix, appPrefixMaxLen, []string{}, appNameMaxRetries)
```

Comment:
we should also consider removing the `existingAppNames` args ( which is set to []string{} ) from the method signature.
As the only other place this is being used also sends a blank list of names https://github.com/redhat-developer/odo/blob/dc489cd720c7d55335e6b3489399b87babf1fe7c/pkg/odo/cli/component/create.go#L189

Reply:
I would prefer retaining the flexibility in generic function